crates/node/src/p2p.rs

```diff
@@ -520,51 +521,82 @@ pub mod testing {
 
     /// A unique seed for each integration test to avoid port conflicts during testing.
     #[derive(Copy, Clone)]
-    pub struct PortSeed(u16);
+    pub struct PortSeed {
+        port_number: u16,
+        case: u16,
+    }
 
     impl PortSeed {
+        // The base port number used, hoping the OS is not using ports in this range
+        pub const BASE_PORT: u16 = 10000;
+        // This constant must be equal to the total number of ports defined below
+        pub const TOTAL_DEFINED_PORTS: u16 = 19;
+        // Maximum number of nodes that can be handled without port collisions
+        pub const MAX_NODES: u16 = 10;
+        // Maximum number of cases that can be handled without port collisions
+        pub const MAX_CASES: u16 = 4;
+        // Each function below corresponds to a port per node. Each defines an offset,
+        // and all offsets must be different
+        pub const TOTAL_PORTS_PER_NODE: u16 = 3;
```

Comment:
changes in this file were needed to be able to use rstest with  different domains and still avoid port collisions